pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -186,13 +186,19 @@ func New(options Options) Index {
 	)...)
 	authzPolicies := kclient.NewDelayedInformer[*securityclient.AuthorizationPolicy](options.Client,
 		gvr.AuthorizationPolicy, kubetypes.StandardInformer, configFilter)
+	// Start with a.stop to ensure the informer respects the index's stop channel
+	authzPolicies.Start(a.stop)
 	AuthzPolicies := krt.WrapClient[*securityclient.AuthorizationPolicy](authzPolicies, opts.WithName("informer/AuthorizationPolicies")...)
 
 	peerAuths := kclient.NewDelayedInformer[*securityclient.PeerAuthentication](options.Client,
 		gvr.PeerAuthentication, kubetypes.StandardInformer, configFilter)
+	// Start with a.stop to ensure the informer respects the index's stop channel
+	peerAuths.Start(a.stop)
```

Comment:
What's the impact (if any) if there are event handlers added after this informer is started? Should these `Start()` calls be in index.Run instead, after all of the setup is complete?

Reply:
Timing doesn't matter. DelayedInformer queues handlers until the CRD is ready, then adds them all before starting the real informer, The .Start() call just stores the stop channel for later.